gateway-ha/src/main/java/io/trino/gateway/ha/handler/QueryIdCachingProxyHandler.java

```diff
@@ -252,13 +285,26 @@ private boolean isPathWhiteListed(String path)
                 || extraWhitelistPaths.stream().anyMatch(s -> path.startsWith(s));
     }
 
-    public boolean handleAuthRequest(HttpServletRequest request)
+    @Override
+    public Optional<Cookie> deleteCookieIfNeeded(HttpServletRequest clientRequest)
     {
-        return true;
+        if (!removeCookiePaths.contains(clientRequest.getRequestURI())) {
+            // TODO: support custom rules for determining if a SessionId cookie should be removed and provide current
+            // path based rule as default option
+            return Optional.empty();
+        }
+        Optional<Cookie> cookie = Arrays.stream(clientRequest.getCookies()).filter(
+                c -> c.getName().equalsIgnoreCase("JSESSIONID")).findAny();
```

Comment:
do we want to set JSESSIONID as static somewhere .. also do we need a separate one to avoid collisions with other cookies potentially?